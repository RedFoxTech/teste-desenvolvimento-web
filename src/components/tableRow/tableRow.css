.pokeList input {
    border: none;
    padding: 2px;
    width: 95px;
    text-align: center;
}

.pokeList input:focus {
    outline: none;
    background: beige;
    border: 2px solid black;
    border-radius: 5px;
    -webkit-box-shadow: 7px 10px 16px -6px rgba(0,0,0,0.75);
    -moz-box-shadow: 7px 10px 16px -6px rgba(0,0,0,0.75);
    box-shadow: 7px 10px 16px -6px rgba(0,0,0,0.75);
}

.pokeIcon {
    width: 30px;
}