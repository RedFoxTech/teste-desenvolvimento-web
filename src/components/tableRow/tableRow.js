import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import './tableRow.css';

const TableRow = (props) => {
    return(
        <>
            <input 
                type="text" 
                value={props.value}
                name={props.name} 
                onChange={(evt) => props.onChangeHandler(evt, props.index)}
            /> 
            {props.src ? 
                <Link to={"/pokedex/"+props.value}>
                    <img 
                        className="pokeIcon" 
                        src={props.src} 
                        alt={props.value}
                        onClick={() => {props.onSendSearch(props.value)}}                  
                    /> 
                </Link>
                : null}
        </>
    );
}

const mapDispatchToProps = dispatch => {
    return {
        onSendSearch: (searching) => dispatch({type: "START_SEARCH", searchParams: searching})
    }
}

export default connect(null, mapDispatchToProps)(TableRow);