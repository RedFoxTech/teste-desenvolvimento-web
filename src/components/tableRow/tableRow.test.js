import React from 'react';
import tableRow from './tableRow';

let wrapper;
beforeEach(() => {
    wrapper = shallow(<tableRow/>);
})

describe("rendering test/snapshot", () => {
    it('should render the proper prototype', () => {
        expect(wrapper).toMatchSnapshot()
    })
})