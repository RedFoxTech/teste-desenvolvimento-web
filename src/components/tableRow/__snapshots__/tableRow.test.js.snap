// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test/snapshot should render the proper prototype 1`] = `<tableRow />`;
